File: apps/junit5-example/src/test/kotlin/example/StepAliasesStoryTest.kt
package example

import dev.executablestories.junit5.Story
import org.junit.jupiter.api.Test
import org.junit.jupiter.api.Assertions.assertEquals
import org.junit.jupiter.api.Assertions.assertTrue

class StepAliasesStoryTest {

    @Test
    fun explicitAndAndButSteps() {
        Story.init("Order with explicit And and But steps")
        Story.given("the user is logged in")
        Story.and("the user has a valid payment method")
        Story.and("the user has items in cart")
        Story.`when`("the user clicks checkout")
        Story.and("confirms the order")
        Story.then("the order should be created")
        Story.and("the payment should be processed")
        Story.but("the inventory is not yet decremented")
        assertTrue(true)
    }

    @Test
    fun butForNegativeAssertion() {
        Story.init("Login blocked for suspended user")
        Story.given("the user account exists")
        Story.given("the account is suspended")
        Story.`when`("the user submits valid credentials")
        Story.then("the user should see an error message")
        Story.but("the user should not be logged in")
        Story.but("the session should not be created")
        assertTrue(true)
    }

    @Test
    fun mixedGivenWhenThenWithAnd() {
        Story.init("Sum calculation using and")
        Story.given("numbers 1, 2, 3, 4, 5")
        Story.and("an accumulator initialized to zero")
        val sum = 1 + 2 + 3 + 4 + 5
        Story.`when`("the sum is calculated")
        Story.then("the result is 15")
        Story.and("the result is positive")
        assertEquals(15, sum)
        assertTrue(sum > 0)
    }
}

File: apps/xunit-example/StepAliasesStoryTest.cs
using ExecutableStories.Xunit;
using Xunit;

namespace XunitExample.Tests;

public class StepAliasesStoryTest
{
    [Fact]
    public void Explicit_And_and_But_steps()
    {
        Story.Init("Order with explicit And and But steps");
        Story.Given("the user is logged in");
        Story.And("the user has a valid payment method");
        Story.And("the user has items in cart");
        Story.When("the user clicks checkout");
        Story.And("confirms the order");
        Story.Then("the order should be created");
        Story.And("the payment should be processed");
        Story.But("the inventory is not yet decremented");
        Story.RecordAndClear();
    }

    [Fact]
    public void But_for_negative_assertion()
    {
        Story.Init("Login blocked for suspended user");
        Story.Given("the user account exists");
        Story.Given("the account is suspended");
        Story.When("the user submits valid credentials");
        Story.Then("the user should see an error message");
        Story.But("the user should not be logged in");
        Story.But("the session should not be created");
        Story.RecordAndClear();
    }

    [Fact]
    public void Mixed_given_when_then_with_And()
    {
        Story.Init("Sum calculation using and");
        Story.Given("numbers 1, 2, 3, 4, 5");
        Story.And("an accumulator initialized to zero");
        var total = 1 + 2 + 3 + 4 + 5;
        Story.When("the sum is calculated");
        Story.Then("the result is 15");
        Story.And("the result is positive");
        Assert.Equal(15, total);
        Assert.True(total > 0);
        Story.RecordAndClear();
    }
}
